src/modrover/learner.py

```diff
@@ -138,25 +112,14 @@ def _initialize_model(self) -> RegmodModel:
         Initialize a regmod model based on the provided modelspecs.
         """
 
-        # Validate that all parameters are represented
-        if not set(self.col_fixed.keys()) == set(self.model_class.param_names):
-            raise ValueError(
-                f"Mismatch between specified parameter names {set(self.col_fixed.keys())} "
-                f"and expected parameter names {set(self.model_class.param_names)}")
-
-        # TODO: Column selection logic should live in the rover class, move out of here
-        # Select the parameter-specific covariate columns
-        all_covariates = {self.col_covs[i - 1] for i in self.cov_ids if i > 0}
-
-        # Add on the fixed columns for every parameter
-        for parameter, columns in self.col_fixed.items():
-            for col in columns:
-                all_covariates.add(col)
-
+        # TODO: this shouldn't be necessary in regmod v1.0.0
+        all_covariates = list(
+            set(itertools.chain.from_iterable(self.col_covs.values()))
+        )
         data = Data(
             col_obs=self.col_obs,
-            col_covs=list(all_covariates),
-            col_offset=self.col_offset,
+            col_covs=all_covariates,
+            col_offset=list(self.col_offset.values())[0],
```

Comment:
This is due to the current `regmod` doesn't allow for different offset columns for different parameters, so when communicate with `regmod` we can only supply one, I just pick the first one. Later when we fix `regmod` this will change